cpp/include/cudf/wrappers/bool.hpp

```diff
@@ -0,0 +1,206 @@
+    return *this;
+  }
+
+  template <typename from_type>
+  CUDA_HOST_DEVICE_CALLABLE bool8& operator=(const from_type& rhs) {
+    this->value = static_cast<uint8_t>(static_cast<bool>(rhs));
+    return *this;
+  }
+
+  // conversion operators
+
+  CUDA_HOST_DEVICE_CALLABLE explicit operator bool() const {
+    return static_cast<bool>(this->value);
+  }
+
+  CUDA_HOST_DEVICE_CALLABLE explicit operator uint8_t() const {
+    return static_cast<uint8_t>(static_cast<bool>(this->value));
+  }
+
+  // ostream << operator overload
+
+  inline std::ostream& operator<<(std::ostream& os) {
+    return os << static_cast<bool>(this->value);
+  }
+
+  // binary operator overloads
+
+  CUDA_HOST_DEVICE_CALLABLE bool operator==(bool8 const &rhs) const {
+    return static_cast<bool>(*this) == static_cast<bool>(rhs);
+  }
+
+  CUDA_HOST_DEVICE_CALLABLE bool operator!=(bool8 const &rhs) const {
+    return static_cast<bool>(*this) != static_cast<bool>(rhs);
+  }
+
+  CUDA_HOST_DEVICE_CALLABLE bool operator<=(bool8 const &rhs) const {
+    return static_cast<bool>(*this) <= static_cast<bool>(rhs); 
+  }
+
+  CUDA_HOST_DEVICE_CALLABLE bool operator>=(bool8 const &rhs) const {
+    return static_cast<bool>(*this) >= static_cast<bool>(rhs); 
+  }
+
+  CUDA_HOST_DEVICE_CALLABLE bool operator<(bool8 const &rhs) const {
+    return static_cast<bool>(*this) < static_cast<bool>(rhs);
+  }
+
+  CUDA_HOST_DEVICE_CALLABLE bool operator>(bool8 const &rhs) const {
+    return static_cast<bool>(*this) > static_cast<bool>(rhs);
+  }
+
+  CUDA_HOST_DEVICE_CALLABLE bool8 operator+(bool8 const &rhs) const {
+    return static_cast<bool8>(static_cast<bool>(*this) +
+                              static_cast<bool>(rhs));
+  }
+
+  CUDA_HOST_DEVICE_CALLABLE bool8 operator-(bool8 const &rhs) const {
+    return static_cast<bool8>(static_cast<bool>(*this) -
+                              static_cast<bool>(rhs));
+  }
+
+  CUDA_HOST_DEVICE_CALLABLE bool8 operator*(bool8 const &rhs) const {
+    return static_cast<bool8>(static_cast<bool>(*this) *
+                              static_cast<bool>(rhs));
+  }
+
+  CUDA_HOST_DEVICE_CALLABLE bool8 operator/(bool8 const &rhs) const {
+    return static_cast<bool8>(static_cast<bool>(*this) /
+                              static_cast<bool>(rhs));
+  }
+
+  // unary operator overloads
+
+  CUDA_HOST_DEVICE_CALLABLE bool8& operator+=(bool8 const &rhs) {
+    bool8 &lhs = *this;
+    lhs = lhs + rhs;
+    return lhs;
+  }
+
+  CUDA_HOST_DEVICE_CALLABLE bool8& operator-=(bool8 const &rhs) {
+    bool8 &lhs = *this;
+    lhs = lhs - rhs;
+    return lhs;
+  }
+
+  CUDA_HOST_DEVICE_CALLABLE bool8& operator*=(bool8 const &rhs) {
+    bool8 &lhs = *this;
+    lhs = lhs * rhs;
+    return lhs;
+  }
+
+  CUDA_HOST_DEVICE_CALLABLE bool8& operator/=(bool8 const &rhs) {
+    bool8 &lhs = *this;
+    lhs = lhs / rhs;
+    return lhs;
+  }
+
+  CUDA_HOST_DEVICE_CALLABLE bool8 operator!() const {
```

Comment:
I think this would be unnecessary if the `operator bool` was implicit? 

I'm trying to think if there would be any negative repercussions of making `operator bool` implicit.

Reply:
I removed the `!` operator overload since now the `bool` conversion is implicit. All tests seem to be passing.